### PR TITLE
issue/cluster-lease-transport-task-ownership

### PR DESCRIPTION
**Pull request by nicole-rappe**
A blackholed PostgreSQL connection could outlive controller or scheduler ownership, while stale scheduler claims could continue dispatching or writing results. Dedicated closable ownership transport, independent expiry guards and holder/generation checks bound recovery and reject stale writes. Durable per-component execution identities prevent blind replay when dispatch acknowledgement is lost.

Refs #466 and umbrella #493. H03 remains OPEN and unmerged through guarded live acceptance; portable validation does not establish five-minute live recovery.

Behavior:
- Lease traffic uses directly closable sockets and an independent watchdog. PostgreSQL17 ownership transactions retain a five-second server timeout, including partitions that drop disconnects. Schema bootstrap and fleet snapshots use the ordinary pool; bootstrap closes before bounded control-pool handoff.
- Claims, heartbeats, dispatch admission, result writes and completion require current holder/generation. Remote work runs after database connections return to pool.
- Durable prepared/dispatching/acknowledged execution records preserve safe replay boundaries. Synchronous patch and maintenance results persist atomically with acknowledgement. Definitive Agent rejection remains Failed; missing/transport-lost replies retain uncertainty. Matching Failed activity results reconcile without changing actual execution status or diagnostics.
- Exactly one matching Running or terminal workflow can settle dispatch while preserving complete workflow and mirrored scheduled-run results. Unacknowledged Pending retains visible uncertainty because persistence precedes executor launch; explicit acknowledgement remains valid. Count all matching identities before checking status so mixed Pending/terminal duplicates remain ambiguous. Unknown outcomes never authorize blind replay.
- Canonical cluster/database/testing guidance and REG056 updated. No schema, dependency or public API change.

Validation at exact 4368645e336e5aa6c8a22fa116add7ec89247a04:
- Full local Engine Go: format/tidy/vet/uncached tests/runtime builds/architecture/204-route inventory; strict docs passed. Real isolated PostgreSQL17 inventory25/zero skips passed, including13 workflow dispatch/recovery subcases. Results timestamp20260906T185903Z. No local race claim.
- [Required PR Validation34053491202](https://github.com/bunny-lab-io/Borealis/actions/runs/34053491202), [CodeQL34053488948](https://github.com/bunny-lab-io/Borealis/actions/runs/34053488948) and container policy34053491220 passed. All portable lanes executed successfully.
- [PostgreSQL artifact9995290218](https://github.com/bunny-lab-io/Borealis/actions/runs/34053491202/artifacts/9995290218) independently audited25 required tests/zero skips,13 workflow subcases; JSONL SHA256981e79be7559891a0a238218feb593c392d84af10dc910343c1a00f3c65dbe1d. Actual Go/PostgreSQL race flags and api-backend/borealis-operator/job-scheduler builds verified in job logs. Ordinary Go database skips do not establish PostgreSQL coverage.
- [Final automated review](https://github.com/bunny-lab-io/Borealis/pull/499#issuecomment-5561460288) completed19:04:40UTC with no major issues; all threads resolved. [Detailed resumable handover](https://github.com/bunny-lab-io/Borealis/issues/466#issuecomment-5561479145) retains commands, artifact paths, historical validation failures and approved integration evidence.

Remaining live gate:
- Deployed /opt/Borealis remains pinned d94708b7220e, with3 K3s nodes Ready, CNPG3/3 and shared storage healthy. Application admission remains1 active/1 desired;02/03 drained. Original operationc13c7396-3e60-45e2-8223-4315a37b85d6 failed attempt4; no retry5/manual hydration/deploy/fault injection.
- Establish immutable rollout/admission sequencing and qualify existing-HMR restoration containing #492 before U01 deployment. Guarded H03 partition acceptance must prove bounded recovery before healing, stale-owner rejection/fencing and healthy3/3 application restoration. Never reboot engine01/manually restart its K3s or roll back joined02/03 snapshots. H04/U03/S01/Q01 remain unfinished; [umbrella register](https://github.com/bunny-lab-io/Borealis/issues/493#issuecomment-5552835571) stays open.

### CODE REVIEW

File: Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_patch_install.go [hunk 1]
```diff
@@ -122,6 +116,9 @@ func (m *goSchedulerManager) runPatchInstallWorkItem(ctx context.Context, item s
 		_ = m.updateScheduledPatchRunStatus(ctx, runID, activityID, scheduledStatusFailed, "", errorText, errorText)
 		return errors.New(errorText)
 	}
+	if err := m.acknowledgeSchedulerDispatch(ctx, requestID); err != nil {
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (nicole-rappe) [reply Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_patch_install.go]: Fixed in 8013393e6445eadba37d216d5e65c4179e7c6b86. Synchronous patch success/failure now writes terminal scheduled-run/activity result and execution acknowledgement in one ownership-checked transaction. Write errors roll back acknowledgement. Required PostgreSQL regression observes the final activity write blocked after earlier writes, cancels it, verifies full rollback, then proves committed results survive takeover without resend or stale overwrite. Local PostgreSQL: 17 tests, zero skips; full Go passed. Fresh branch CI requested; combined-main validation remains after conflict checkpoint.

File: Data/Engine/Containers/api-backend/cmd/api-backend/postgres_control_connector.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	if strings.HasPrefix(dsn, "postgres://") || strings.HasPrefix(dsn, "postgresql://") {
+		var err error
+		dsn, err = pq.ParseURL(dsn)
+		if err != nil {
+			return nil, err
+		}
+	}
+	// PostgreSQL 17 terminates the entire transaction, including idle periods
+	// and lock waits. Enforce this per control connection, never cluster-wide.
+	return pq.NewConnector(dsn + " transaction_timeout=5000")
```
[the remark this answers is not in the document]
Comment (nicole-rappe) [reply Data/Engine/Containers/api-backend/cmd/api-backend/postgres_control_connector.go:29]: Fixed in 8013393e6445eadba37d216d5e65c4179e7c6b86. Scheduler ordinary store (including fleet snapshots/bootstrap/health) uses normal PostgreSQL pool; a separate bounded pool owns claims, recovery, dispatch/result transactions and completion. Lease transport remains separate. Real PostgreSQL test blocks an actual two-target snapshot for over five seconds, then verifies both targets commit while ownership pool retains transaction_timeout=5s. Local PostgreSQL: 17 tests, zero skips; full Go passed. Fresh branch CI requested.

**Comment by nicole-rappe** [comment]
@codex review

Please re-review H03 at `8013393e6445eadba37d216d5e65c4179e7c6b86`, especially both P1 fixes: atomic synchronous patch result/acknowledgement and separate ownership pool preserving fleet snapshot transaction limits. Local full Go, PostgreSQL (17 required tests, zero skips), K3s/architecture and strict docs passed. Branch workflow_dispatch requests full portable validation despite main conflicts.

PR remains open pending refreshed-main integration and live three-node acceptance. Five documentation/generated-inventory conflicts are inspected; no resolution applied while separate AGENTS.md conflict checkpoint awaits operator. Existing general merge approval persists, but no merge/qualification claim is made.

File: Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_manager.go [hunk 3]
```diff
@@ -954,13 +964,17 @@ func (m *goSchedulerManager) runAgentMaintenanceWorkItem(ctx context.Context, it
...
 			}
+			if err := m.beginSchedulerDispatch(ctx, operationID); err != nil {
+				return err
+			}
 			response, lastState, lastErr = m.callSiteWorkerHostService(ctx, snapshot.Route, body, callTimeout+2*time.Second)
 			if lastErr == nil {
+				if err := m.acknowledgeSchedulerDispatch(ctx, operationID); err != nil {
+					return err
+				}
 				break
```
[the remark this answers is not in the document]
Comment (nicole-rappe) [reply Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_manager.go]: Fixed in 0792da6799feae3553032d84885e208c5f90f067. Agent maintenance now commits acknowledgement, response-derived Running/Skipped run/activity status and diagnostics in one ownership-checked transaction. Final-write cancellation rolls everything back. Required PostgreSQL regression covers both responses, actual final-write lock/cancellation, committed takeover without resend and stale generation rejection. Local full Go, strict docs and PostgreSQL passed: 18 required tests, zero skips. Fresh full CI requested.

**Comment by nicole-rappe** [comment]
@codex review

Please re-review final head `0792da6799feae3553032d84885e208c5f90f067` including maintenance response/acknowledgement atomicity follow-up, original patch result atomicity and bounded ownership-pool fixes. Local full Go/docs and all 18 PostgreSQL tests passed without skips. Workflow now builds all images for manual full validation; prior corrected full CI 34005537970 passed and logs verified all nine services (WebUI dev/prod targets). Main conflicts and live three-node gate remain open.

File: Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_manager.go [hunk 4]
```diff
@@ -954,13 +964,14 @@ func (m *goSchedulerManager) runAgentMaintenanceWorkItem(ctx context.Context, it
...
+				return err
+			}
 			response, lastState, lastErr = m.callSiteWorkerHostService(ctx, snapshot.Route, body, callTimeout+2*time.Second)
 			if lastErr == nil {
 				break
 			}
-			if lastState == "agent_error" || lastState == "invalid_agent_response" {
-				break
-			}
+			return fmt.Errorf("%w: %s: %v", errSchedulerOutcomeUnknown, operationID, lastErr)
```
[the remark this answers is not in the document]
Comment (nicole-rappe) [reply Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_manager.go:974]: Fixed in b1ffeb34c1ac5d5bcf1eaf5af6ff47df9cc1480d. Actual Agent status=error responses now persist Failed result and acknowledgement atomically for maintenance and patch dispatch. Worker error labels without Agent response, transport loss and missing replies stay uncertain. Maintenance acceptance also preserves an already-committed terminal callback instead of resetting it to Running. HTTP classification regression plus required PostgreSQL Failed/early-completion/rollback/takeover coverage passed; local full Go, strict docs and all 18 PostgreSQL tests passed without skips (20260906T022201Z).

**Comment by nicole-rappe** [comment]
@codex review

Please re-review `b1ffeb34c1ac5d5bcf1eaf5af6ff47df9cc1480d`. Explicit Agent rejection now commits Failed result/acknowledgement for both maintenance and patch dispatch; worker/missing/transport outcomes remain unknown. Fast terminal callback before acceptance remains terminal. Local full Go/docs and all 18 PostgreSQL tests passed, including actual HTTP response classification and transaction rollback/takeover. Fresh full-image CI requested. Prior final runtime head 0792da679 passed full CI 34005849825 with audited 18-test race artifact and all nine service image builds. Main conflicts and live three-node acceptance remain separate gates.

File: .github/workflows/pr-validation.yml [hunk 5]
```diff
@@ -260,7 +263,13 @@ jobs:
         with:
           go-version: 1.25.12
           cache-dependency-path: Data/Engine/Containers/api-backend/go.sum
-      - run: ./Tests/run-containers.sh --base "$BOREALIS_CI_BASE" --head "$BOREALIS_CI_HEAD"
+      - name: Build selected images
+        run: |
+          if [[ "$BOREALIS_CI_FORCE_ALL" == "true" ]]; then
+            ./Tests/run-containers.sh --all
```
[the remark this answers is not in the document]
Comment (nicole-rappe) [reply .github/workflows/pr-validation.yml:272]: No code change needed for this finding. Tests/helpers/changed_paths.py:54 checks BOREALIS_CI_FORCE_ALL=true and returns true before reading base/head, so workflow_dispatch enters the container job. Exact-head manual CI34006362976 at b1ffeb34c passed; job101414256867 log proves all nine service image builds completed, including WebUI dev/prod targets. https://github.com/bunny-lab-io/Borealis/actions/runs/34006362976/job/101414256867 . Required database artifact9981076906 also audited:18 tests, zero skips with race detector. Job-level skip described here did not occur.

**Comment by nicole-rappe** [comment]
@codex review

Main integrated at 6b822b082c19e1348e1c9e33a3a518a2cafc029d using approved five-file documentation/inventory resolution. Inspect H03 ownership changes alongside H02 admission reconciliation, U01 HMR recovery gates, H05 TLS lifecycle and H06 release identity. Local full Engine Go, strict docs, K3s policy and all24 required PostgreSQL tests passed with zero skips; fresh PR CI34032207255 running. All previous review threads resolved on prior head; this requests fresh integration review.

**Comment by nicole-rappe** [comment]
@codex review

Main integration now includes separately reviewed/merged #510 token-permission correction. Conflict-free integration; previous H03 ownership logic unchanged. Review current head and retain existing live acceptance gates. Fresh CI running; prior exact6b822b082 evidence retained in PR body.

File: Data/Engine/Containers/api-backend/cmd/api-backend/cluster_controller.go [hunk 6]
```diff
@@ -251,7 +288,7 @@ func clusterControllerMode() bool {
 }
 
 func runClusterController(ctx context.Context, cfg gatewayConfig) error {
-	operatorStore, closeStore, err := openOperatorStore(cfg)
+	operatorStore, closeStore, err := openControlOperatorStore(cfg)
```
[the remark this answers is not in the document]
Comment (nicole-rappe) [reply Data/Engine/Containers/api-backend/cmd/api-backend/cluster_controller.go:291]: Addressed in155104a37. Bootstrap completes through ordinary pool using existing deadline, which closes before bounded controller pool handoff. Real PostgreSQL test holds assembly DDL lock beyond5s and verifies successful startup, retained5s ownership limit, and closed bootstrap backend. Local PostgreSQL17 lane:25 required/zero skips; full Engine Go and strict docs passed. Fresh CI/race and review requested.

**Comment by nicole-rappe** [comment]
@codex review

Please review current155104a37 including bootstrap-pool correction to3944056033 and regression test. Local real PostgreSQL17 required inventory25/zero skips, full Engine Go and strict docs passed. Live qualification remains outstanding; no merge yet.

File: Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_execution_ownership.go [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+		return false, err
+	}
+	if record.State == "dispatching" && record.ActivityID > 0 {
+		var state string
+		err := tx.QueryRowContext(ctx, `SELECT status FROM engine.activity_history
+			WHERE id=$1 AND metadata_json::jsonb->>'scheduler_execution_id'=$2`, record.ActivityID, record.ID).Scan(&state)
+		if err != nil && !errors.Is(err, sql.ErrNoRows) {
+			return false, err
+		}
+		if err == nil && stringInSet(strings.ToLower(state), "success", "completed", "succeeded") {
```
[the remark this answers is not in the document]
Comment (nicole-rappe) [reply Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_execution_ownership.go]: Preserved draft now passes actual PostgreSQL17 inventory25/zero skips including Success/Failed subcases, with status/diagnostic preservation and mismatched-identity rejection. Canonical ownership notes/REG056 updated. Local clean commit6c7b26d456ceb0a2262312e864cdd8d72866a491 remains intentionally unpushed until F03/F04 prerequisite integration; no new CI claim. Full diff and resume evidence: https://github.com/bunny-lab-io/Borealis/issues/466#issuecomment-5561134798 . Thread remains open until pushed-head validation.

**Comment by nicole-rappe** [comment]
@codex review

Review exact9bb02f1700d17df458b6cebea623d28fda0d85bf: preserved Failed-result correction from6c7b26d45 now pushed, plus one integration of independently reviewed/merged F02/F03/F04 prerequisites. Prior review completed17:20:47UTC; no duplicate review running. Approved registry union preserves REG056/057/058; preparation source/tests exactly match reviewed main, scheduler correction unchanged during integration.

Local full Engine Go (vet, tests, runtime builds,204-route contract), strict docs and references passed on integrated tree. Preserved ownership correction passed actual PostgreSQL17 inventory25/zero skips including Success/Failed subcases before integration; exact-head CI34052251444 now supplies required race/PostgreSQL evidence. Previous overall155104a37 CI remains failed and is not relabeled; enrollment correction is now integrated.

Current Task ~97%; Current Stage ~70%; Overall Task ~56%. Remaining gates: current CI/race artifact audit/review, guarded H03 live acceptance and unresolved existing-HMR sequencing. H03 must stay OPEN/unmerged through live acceptance; safe checkpoint before next live retry.

File: Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_execution_ownership.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (nicole-rappe) [reply Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_execution_ownership.go]: Pushed and independently validated on9bb02f1700d17df458b6cebea623d28fda0d85bf. Full PR Validation34052251444 and CodeQL34052248112 passed. Downloaded database artifact9994923427 and reran inventory auditor:25 required tests/zero skips, both Success and Failed subcases pass; definitive result/diagnostic preservation and unrelated-identity rejection retained. Actual job101537882935 logs prove CGO_ENABLED=1 GOFLAGS=-race; Engine Go job101537882922 proves race enabled too. Artifact: https://github.com/bunny-lab-io/Borealis/actions/runs/34052251444/artifacts/9994923427 . JSONL SHA2567cff40e91f3c932a1e4d31fac357f019e876851484a29043591384f3f3205109. Resolving addressed finding; fresh overall review still running, and guarded live acceptance remains open.

**Comment by nicole-rappe** [comment]
@codex review

Review correction at exact f7eb1c0f409d8de8bae9691df6280a6d07fb553e, focused on prior finding https://github.com/bunny-lab-io/Borealis/pull/499#discussion_r3944906260 . Previous review completed; no duplicate request outstanding.

Workflow start commits its row before replying and executes asynchronously. Recovery now accepts exactly one matching scheduler_execution_id in any persisted workflow status, retains run ID and skips replay. Missing/unrelated/duplicate identities remain uncertain. Existing required PostgreSQL workflow test now covers outgoing identity plus seven statuses and four negative identity cases; expiry/reclaim, stale acknowledgement rejection, blocked replay, and full saved workflow/mirrored scheduled result preservation asserted. Canonical ownership docs and REG056 updated.

Local full Engine Go and strict docs passed. Isolated PostgreSQL17 lane passed25 required tests/zero skips, including12 workflow subcases. Results all timestamp20260906T184946Z. Exact-head PR Validation34053034871 and CodeQL34053032366 running; prior9bb02f170 CI success remains scoped to prior head.

Current Task ~98%; Current Stage ~82%; Overall Task ~56%. Remaining gates: exact-head CI/artifact audit/review, guarded H03 live acceptance and existing-HMR sequencing. H03 stays OPEN/unmerged. Safe pause before live retry; runtime remains pinned d947 with no runtime changes.

File: Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_execution_ownership.go [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+			record.State = "acknowledged"
+			if err := storeSchedulerExecution(ctx, tx, item, record); err != nil {
+				return false, err
+			}
+		}
+	}
+	if record.State == "dispatching" && item.Kind == schedulerKindScheduledWorkflowRun {
+		var count int
+		var runID sql.NullInt64
+		if err := tx.QueryRowContext(ctx, `SELECT COUNT(*),MIN(id) FROM engine.workflow_runs WHERE source_metadata_json::jsonb->>'scheduler_execution_id'=$1 AND LOWER(status) IN ('running','success','completed','succeeded')`, record.ID).Scan(&count, &runID); err != nil {
```
[the remark this answers is not in the document]
Comment (nicole-rappe) [reply Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_execution_ownership.go]: Fixed in pushed f7eb1c0f409d8de8bae9691df6280a6d07fb553e. Recovery requires exactly one matching persisted workflow execution ID regardless of status. Run ID becomes acknowledged without replay or result writes. Missing/unrelated/duplicate identities remain uncertain.

Existing required PostgreSQL workflow regression now covers outgoing identity plus all seven statuses and four identity guards, expired claim takeover, stale acknowledgement rejection, blocked replay, and complete workflow/mirrored scheduled row preservation. Canonical ownership docs and REG056 updated.

Exact-head CI34053034871 and CodeQL34053032366 passed. Downloaded database artifact9995154112 independently audited25 required tests/zero skips and12 workflow subcases; job101539974861 proves CGO_ENABLED=1 and GOFLAGS=-race. JSONL SHA2568989aa4a8c77be936a817fd7cbcfe5f2d8d7b20cc0df5a8a306998a4e4f93e8c. Full local Go/docs and isolated PostgreSQL also passed. Fresh focused review remains pending; no live qualification claim.

Current Task ~99%; Current Stage ~84%; Overall Task ~56%. Remaining: fresh review and guarded live acceptance. Runtime remains pinned; H03 stays OPEN/unmerged at safe checkpoint.

**Comment by nicole-rappe** [comment]
@codex review

Focused follow-up at exact 4368645e336e5aa6c8a22fa116add7ec89247a04 for https://github.com/bunny-lab-io/Borealis/pull/499#discussion_r3944937911 . Prior review completed18:55:24UTC; new Pending-launch finding required this correction. Please check changed recovery predicate and regression delta; previously reviewed prerequisite integrations unchanged.

Unacknowledged Pending now retains visible outcome_unknown because persistence precedes executor launch. Exactly one matching Running or terminal workflow can settle dispatch; count all matching identities before checking status so mixed Pending/terminal duplicates remain ambiguous. Explicit acknowledgement remains accepted while Pending. No replay or fabricated terminal result introduced.

Local full Go and strict docs passed; isolated PostgreSQL17 passed25 required/zero skips with13 workflow subcases, including unacknowledged/acknowledged Pending, preserved results for accepted recovery, mixed-status duplicates, takeover and stale-ack rejection. Results timestamp20260906T185903Z; JSONL SHA256831d756aa1dacf2e21a97ee7396781b3e6171e4037b60bbfd82c40a6e1d73ea8. Current CI34053491202 / CodeQL34053488948 running. Priorf7eb1c0f4 CI passed but exposed Pending behavior is superseded by this correction.

Current Task ~99%; Current Stage ~84%; Overall Task ~56%. Remaining gates: exact-head CI/artifact/review, guarded H03 live acceptance and existing-HMR sequencing. Runtime remains clean/pinned d947, healthy K3s/CNPG/storage, app1/1. H03/493 stay OPEN; safe pause before live retry.

File: Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_execution_ownership.go [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+				return false, err
+			}
+		}
+	}
+	if record.State == "dispatching" && item.Kind == schedulerKindScheduledWorkflowRun {
+		var count int
+		var runID sql.NullInt64
+		// Workflow start persists its run before replying and executes asynchronously.
+		// Any status proves acceptance; multiple matching rows remain ambiguous.
+		if err := tx.QueryRowContext(ctx, `SELECT COUNT(*),MIN(id) FROM engine.workflow_runs WHERE source_metadata_json::jsonb->>'scheduler_execution_id'=$1`, record.ID).Scan(&count, &runID); err != nil {
```
[the remark this answers is not in the document]
Comment (nicole-rappe) [reply Data/Engine/Containers/api-backend/cmd/api-backend/scheduler_execution_ownership.go]: Addressed in4368645e336e5aa6c8a22fa116add7ec89247a04. Unacknowledged Pending retains outcome_unknown, stays unreplayed and keeps workflow Pending/scheduled run Running. Explicit acknowledgement still wins. Running/terminal matching evidence can reconcile only after counting every identity match; mixed Pending/terminal duplicates remain ambiguous.

Regression now has13 workflow subcases, including both Pending boundaries, mixed-status duplicates, expiry/takeover, stale acknowledgement rejection, replay rejection and complete saved result preservation. Canonical ownership docs and REG056 updated.

Exact-head PR Validation34053491202 and CodeQL34053488948 passed. Artifact9995290218 independently audited25 required PostgreSQL tests/zero skips and13 workflow subcases; database job101541211379 proves CGO_ENABLED=1/GOFLAGS=-race, Go job101541211383 proves race lane, image job101541211388 proves actual api-backend/operator/scheduler builds. Audited JSONL SHA256981e79be7559891a0a238218feb593c392d84af10dc910343c1a00f3c65dbe1d. Full local Go/docs/isolated PostgreSQL also passed. Fresh focused review pending; no live acceptance claim.

Current Task ~99%; Current Stage ~84%; Overall Task ~56%. Review and guarded live gates remain. Runtime stays pinned; safe pause before live retry.
